### PR TITLE
LPS-185674 Add Relationship, Position, and Linked Features to Model Builder

### PR DESCRIPTION
**Pull request by igor-franca**

### CODE REVIEW

File: modules/apps/object/object-js-components-web/src/main/resources/META-INF/resources/utils/api.ts [hunk 1]
```diff
@@ -194,7 +190,7 @@ export async function getAllObjectFolders() {
 	);
 }
 
-export async function getFolderByERC(folderERC: string) {
+export async function getObjectFolderByERC(folderERC: string) {
```
Comment (naraandrade) [inline modules/apps/object/object-js-components-web/src/main/resources/META-INF/resources/utils/api.ts:193]: @igor-franca Please, sort it in alphabetical order. Remember to regen the .d.ts file.

File: modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ModelBuilder/ModelBuilderContext/objectFolderReducer.ts [hunk 2]
```diff
@@ -605,22 +616,25 @@ export function ObjectFolderReducer(state: TState, action: TAction) {
...
+						...objectDefinitionNode.data,
+						nodeSelected:
+							objectDefinitionNode.id ===
+							selectedObjectDefinitionId,
+					},
+				})
+			);
 
 			const newLeftSidebarItems = leftSidebarItems.map((sidebarItem) => {
 				const newLeftSidebarDefinitions = sidebarItem.objectDefinitions?.map(
```
Comment (naraandrade) [inline modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ModelBuilder/ModelBuilderContext/objectFolderReducer.ts:632]: @igor-franca I think this should be `newLeftSidebarObjectDefinitions`

File: modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ModelBuilder/ModelBuilderContext/objectFolderReducer.ts [hunk 3]
```diff
@@ -605,22 +616,25 @@ export function ObjectFolderReducer(state: TState, action: TAction) {
...
 					(sidebarDefinition) => ({
```
Comment (naraandrade) [inline modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ModelBuilder/ModelBuilderContext/objectFolderReducer.ts:633]: @igor-franca `sidebarObjectDefinition`

File: modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ViewObjectDefinitions/ViewObjectDefinitions.tsx [hunk 4]
```diff
@@ -414,49 +414,49 @@ export default function ViewObjectDefinitions({
...
 
-			{showModal.editFolder && (
-				<ModalEditFolder
-					externalReferenceCode={
-						selectedFolder.externalReferenceCode as string
-					}
-					folderID={selectedFolder.id as number}
+			{showModal.deleteObjectFolder && (
+				<ModalDeleteObjectFolder
+					folder={selectedFolder as ObjectFolder}
```
Comment (naraandrade) [inline modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ViewObjectDefinitions/ViewObjectDefinitions.tsx:432]: @igor-franca `selectedObjectFolder`

**Review by naraandrade — CHANGES_REQUESTED** [review]

**Review by naraandrade — APPROVED** [review]

**Comment by igor-franca** [comment]
ci:forward:force

**Comment by naraandrade** [comment]
Just started reviewing :)

**Comment by naraandrade** [comment]
ci:forward:force

**Pull request closed without merging**